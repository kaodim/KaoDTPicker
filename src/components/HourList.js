import React, { PropTypes } from 'react'
import HourListItem from './HourListItem'

const HourList = ({ onChange, selectedTime, timeslots }) => {
  const slots = {
    morning: [],
    afternoon: [],
    evening: [],
    night: []
  }
  // Logic for spliting morning, afternoon and etc
  timeslots.map((ts) => {
    const time = new Date(ts.value)
    const hour = time.getHours()
    const minute = time.getMinutes()
    const minuteRep = minute === 30 ? 0.5 : 0
    const timeTotal = hour + minuteRep
    if (timeTotal <= 12) {
      slots.morning.push(ts)
    } else if (timeTotal <= 17) {
      slots.afternoon.push(ts)
    } else if (timeTotal <= 20) {
      slots.evening.push(ts)
    } else if (timeTotal <= 23) {
      slots.night.push(ts)
    }
  })
  return (
    <div className="kld-hourlist">
      {Object.keys(slots).map((zoneDay, index) => {
        return (
          <HourListItem
            key={index}
            label={zoneDay.toUpperCase()}
            onChangeTime={onChange}
            selectedTime={selectedTime}
            timeslots={slots[zoneDay]}
          />
        )
      })}
    </div>
  )
}
HourList.defaultProps = {
  onChange: () => {},
  selectedTime: '',
  timeslots: []
}

HourList.propTypes = {
  onChange: PropTypes.func,
  selectedTime: PropTypes.string,
  timeslots: PropTypes.array
}

export default HourList
