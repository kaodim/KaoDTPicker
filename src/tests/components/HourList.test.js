import React from 'react'
import { shallow } from 'enzyme'
import HourList from '../../components/HourList'
import calTimeslots from '../fixtures/calTimeslots'

let wrapper

beforeEach(() => {
  wrapper = shallow(<HourList timeslots={calTimeslots} />)
})

test('should render HourList correctly', () => {
  // expect(wrapper).toMatchSnapshot()
})

test('should render correct amount of HourListItem', () => {
  const lenghtOfHourListItem = calTimeslots.length
  expect(wrapper.find('HourListItem').length).toBe(lenghtOfHourListItem)
})

test('should render HourList with selected date', () => {
  const now = '2020-01-17T07:00:00.000+08:00'
  const onChangeSpy = jest.fn()
  wrapper.setProps({
    selectedTime: now,
    onChange: onChangeSpy(now)
  })
  expect(onChangeSpy).toHaveBeenLastCalledWith(now)
  // expect(wrapper).toMatchSnapshot()
})
