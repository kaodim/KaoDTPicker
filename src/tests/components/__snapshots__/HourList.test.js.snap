// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HourList correctly 1`] = `
<div
  className="kld-hourlist"
>
  <HourListItem
    key="0"
    label="MORNING"
    onChangeTime={[Function]}
    selectedTime=""
    timeslots={
      Array [
        Object {
          "available": true,
          "human_readable_price": "",
          "localized_total_price": "RM50",
          "price": 0,
          "show_price": false,
          "surchargable": false,
          "total_price": 50,
          "value": "2020-01-17T07:00:00.000+08:00",
        },
      ]
    }
  />
  <HourListItem
    key="1"
    label="AFTERNOON"
    onChangeTime={[Function]}
    selectedTime=""
    timeslots={
      Array [
        Object {
          "available": true,
          "human_readable_price": "",
          "localized_total_price": "RM50",
          "price": 0,
          "show_price": false,
          "surchargable": false,
          "total_price": 50,
          "value": "2020-01-17T12:30:00.000+08:00",
        },
      ]
    }
  />
  <HourListItem
    key="2"
    label="EVENING"
    onChangeTime={[Function]}
    selectedTime=""
    timeslots={
      Array [
        Object {
          "available": true,
          "human_readable_price": "",
          "localized_total_price": "RM50",
          "price": 0,
          "show_price": false,
          "surchargable": false,
          "total_price": 50,
          "value": "2020-01-17T17:30:00.000+08:00",
        },
      ]
    }
  />
</div>
`;
